### PR TITLE
Add an option to RazorPrecompileModule to allow conditional compilation

### LINKED ISSUE #3276: Conditionally enable precompilation depending on project configuration
A workaround to making the dev experience tolerable knowing https://github.com/aspnet/Mvc/issues/3244 is problematic. We'll add support for conditionally precompiling depending on configuration with a way for users to override this.

### PR DESCRIPTION
**Pull request by pranavkm**
based on Configuration.

Fixes #3276

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @Eilon

File: src/Microsoft.AspNet.Mvc.Razor/Precompilation/RazorPreCompileModule.cs [hunk 1]
```diff
@@ -67,6 +73,28 @@ public void AfterCompile(AfterCompileContext context)
...
+
+        /// <summary>
+        /// Determines if this instance of <see cref="RazorPreCompileModule"/> should enable
+        /// compilation of views.
+        /// </summary>
+        /// <param name="context">The <see cref="BeforeCompileContext"/>.</param>
+        /// <returns><c>true</c> if views should be precompiled; otherwise <c>false</c>.</returns>
+        /// <remarks>Returns <c>true</c> if the current application is being built in <c>release</c>
+        /// configuration.</remarks>
+        protected virtual bool EnablePreCompilation(BeforeCompileContext context)
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Razor/Precompilation/RazorPreCompileModule.cs]: We have a property on this type `GenerateSymbols` which could deserve the same treatment - i.e. change it to a method which accept the context. Although I don't see why you would need symbols for precompiled modules.

File: src/Microsoft.AspNet.Mvc.Razor/Precompilation/RazorPreCompileModule.cs [hunk 2]
```diff
@@ -67,6 +73,28 @@ public void AfterCompile(AfterCompileContext context)
         {
         }
 
+
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Razor/Precompilation/RazorPreCompileModule.cs]: extra white space

**Comment by Eilon** [comment]
:shipit:

**Pull request closed without merging**